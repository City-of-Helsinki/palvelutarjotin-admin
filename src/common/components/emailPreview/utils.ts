import { TEMPLATE_TYPE } from '../../../domain/enrolment/types';
import { EmailTemplatesQuery } from '../../../generated/graphql';
import { Language } from '../../../types';

export const getTranslatedTemplate = (
  templates: EmailTemplatesQuery | undefined,
  templateType: TEMPLATE_TYPE,
  locale: Language
) => {
  if (!templates) return null;

  const template = templates.notificationTemplates?.edges.find(
    (t) => t?.node?.type === templateType
  );
  const translatedTemplate = template?.node?.translations.find(
    (i) => i?.languageCode.toLowerCase() === locale
  );

  return translatedTemplate;
};
