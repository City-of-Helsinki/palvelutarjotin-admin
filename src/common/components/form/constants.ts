export const invalidFieldClass = 'invalid-field';
