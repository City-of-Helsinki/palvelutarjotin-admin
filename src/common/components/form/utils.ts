/* eslint-disable @typescript-eslint/no-explicit-any */
import { FormikErrors, FormikTouched } from 'formik';
import { TFunction } from 'i18next';
import get from 'lodash/get';

/** Get error text
 * @param {Object} errors
 * @param {Object} touched
 * @param {string} name
 * @param {Function} t
 * @return {string}
 */
export const getErrorText = (
  errors: FormikErrors<any>,
  touched: FormikTouched<any>,
  name: string,
  t: TFunction
) => {
  const error: any = get(errors, name);

  return !!error && get(touched, name)
    ? typeof error === 'string'
      ? (t(error) as string)
      : (t(error.key, error) as unknown as string)
    : '';
};
