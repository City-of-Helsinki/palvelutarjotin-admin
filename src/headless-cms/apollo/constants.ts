import AppConfig from '../../domain/app/AppConfig';

export const persistorLocalStorageKey = `${AppConfig.appName}-cms-${AppConfig.environment}-apollo-cache`;
// eslint-disable-next-line max-len
export const timePersistedLocalStorageKey = `${AppConfig.appName}-cms-${AppConfig.environment}-apollo-cache-persisted-at`;
