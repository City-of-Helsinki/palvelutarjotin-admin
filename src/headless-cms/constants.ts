import type { Language } from '../types';

export const HEADER_MENU_NAME = {
  en: 'Palvelutarjotin all UIs Header (EN)',
  fi: 'Palvelutarjotin all UIs Header (FI)',
  sv: 'Palvelutarjotin all UIs Header (SV)',
} as const satisfies Record<Language, string>;

export const FOOTER_MENU_NAME = {
  en: 'Palvelutarjotin-admin Footer (EN)',
  fi: 'Palvelutarjotin-admin Footer (FI)',
  sv: 'Palvelutarjotin-admin Footer (SV)',
} as const satisfies Record<Language, string>;
