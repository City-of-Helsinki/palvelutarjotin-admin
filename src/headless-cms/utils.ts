// '/segment1/segment2/' -> ['/segment1/', '/segment1/segment2/']
// current implementation required both leading and trailing slashes
// to include all breadcrumbs
export const uriToBreadcrumbs = (uri: string): string[] => {
  return slugsToUriSegments(
    stripLocaleFromUri(uri)
      .split('/')
      // Filter out empty strings
      .filter((i) => i)
  );
};

export const slugsToUriSegments = (slugs: string[]): string[] => {
  return slugs.map((slug, index) => {
    return `/${slugs.slice(0, index + 1).join('/')}/`;
  });
};

export const stripLocaleFromUri = (uri: string): string => {
  return uri.replace(/^\/(en|sv|fi)(?![a-z0-9])/i, '');
};

export const getCmsUriFromPath = (path: string) => {
  return path.replace(/(\/?(fi|en|sv))?\/?cms-page/, '');
};

export const removeSurroundingSlashes = (path: string) => {
  return path.replace(/^\/|\/$/g, '');
};
