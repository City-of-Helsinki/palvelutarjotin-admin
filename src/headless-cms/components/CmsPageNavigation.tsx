import React from 'react';
import { Link } from 'react-router-dom';

import { ROUTES } from '../../domain/app/routes/constants';
import { Page } from '../../generated/graphql-cms';

const CmsPageNavigation: React.FC<{
  page: Page | undefined | null;
}> = ({ page }): JSX.Element => {
  const parentPage = page?.parent?.node as Page;
  const childrenPages = (page?.children?.nodes as Page[]) ?? [];

  return (
    <div>
      <p>CmsPageNavigation {page?.id}</p>
      Parent:
      <ul>
        {parentPage && (
          <li>
            <Link
              to={`${ROUTES.CMS_PAGE.replace(
                ':id',
                parentPage.translation?.id as string
              )}`}
            >
              {parentPage.translation?.title}
            </Link>
          </li>
        )}
      </ul>
      Children:
      <ul>
        {childrenPages
          .filter((subPage) => subPage != null)
          .map((subPage) => (
            <li>
              <Link
                to={`${ROUTES.CMS_PAGE.replace(
                  ':id',
                  subPage?.translation?.id as string
                )}`}
              >
                {subPage?.translation?.title}
              </Link>
            </li>
          ))}
      </ul>
    </div>
  );
};

export default CmsPageNavigation;
