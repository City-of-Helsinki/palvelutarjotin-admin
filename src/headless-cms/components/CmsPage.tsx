import React from 'react';
import { useParams } from 'react-router';

import LoadingSpinner from '../../common/components/loadingSpinner/LoadingSpinner';
import {
  LanguageCodeEnum,
  Page,
  usePageQuery,
} from '../../generated/graphql-cms';
import useLocale from '../../hooks/useLocale';
import apolloClient from '../client';
import CmsPageContent from './CmsPageContent';
import CmsPageNavigation from './CmsPageNavigation';

const CmsPage: React.FC = () => {
  const locale = useLocale();
  const { id: pageId } = useParams<{ id: string }>();
  const { data: pageData, loading } = usePageQuery({
    client: apolloClient,
    variables: {
      id: pageId,
      language: locale.toString().toUpperCase() as LanguageCodeEnum,
    },
  });

  return (
    <LoadingSpinner isLoading={loading}>
      <CmsPageNavigation page={pageData?.page as Page} />
      <CmsPageContent page={pageData?.page as Page} />
    </LoadingSpinner>
  );
};

export default CmsPage;
