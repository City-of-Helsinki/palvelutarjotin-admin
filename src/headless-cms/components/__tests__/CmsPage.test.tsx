import { dequal } from 'dequal';
import { graphql } from 'msw';
import React from 'react';

import { ROUTES } from '../../../domain/app/routes/constants';
import { PageIdType } from '../../../generated/graphql-cms';
import { server } from '../../../test/msw/server';
import { fakePage } from '../../../utils/cmsMockDataUtils';
import {
  act,
  renderWithRoute,
  screen,
  userEvent,
  waitFor,
  within,
} from '../../../utils/testUtils';
import { normalizeCmsUri } from '../../utils';
import CmsPage, { breadcrumbsContainerTestId } from '../CmsPage';
import { cmsNavigationContainerTestId } from '../CmsPageNavigation';

const wait = () => act(() => new Promise((res) => setTimeout(res, 500)));

type PageHierarchy = {
  title: string;
  uri: string;
  content?: string;
  children?: PageHierarchy[];
};

const mainPageTitle = 'Pääsivun aliotsikko';
const mainPageContent = 'Pääsivun kontentti';
const subPageTitle = 'Alisivun otsikko';
const subPageContent = 'Alisivun kontentti';

const searchablePages = [
  {
    title: 'Oppimateriaali1',
    uri: '/paasivu/oppimateriaalit/oppimateriaali1/',
  },
  {
    title: 'Oppimateriaali2',
    uri: '/paasivu/oppimateriaalit/oppimateriaali2/',
  },
  {
    title: 'Oppimateriaali3',
    uri: '/paasivu/oppimateriaalit/oppimateriaali3/',
  },
  {
    title: 'Oppimateriaali4',
    uri: '/paasivu/oppimateriaalit/oppimateriaali4/',
  },
  {
    title: 'Oppimateriaali5',
    uri: '/paasivu/oppimateriaalit/oppimateriaali5/',
  },
  {
    title: 'Oppimateriaali6',
    uri: '/paasivu/oppimateriaalit/oppimateriaali6/',
  },
];

const learningMaterialsPage = {
  title: 'Oppimateriaalit',
  uri: '/paasivu/oppimateriaalit/',
  content: `<p>Kaikki opimateriaalit</p>`,
  children: searchablePages,
};

const pageHierarchy: PageHierarchy[] = [
  {
    title: 'Pääsivu',
    uri: '/paasivu/',
    content: `<h2>${mainPageTitle}</h2><p>${mainPageContent}</p>`,
    children: [
      {
        title: 'Alisivu1',
        uri: '/paasivu/alisivu1/',
        content: `<h2>${subPageTitle}</h2><p>${subPageContent}</p>`,
        children: [
          {
            title: 'Alisivu1 alisivu1',
            uri: '/paasivu/alisivu1/ali2sivu1/',
            content: '<p>Sisältö</p>',
            children: [
              {
                title: 'Alisivu1 alisivu1 alisivu1',
                uri: '/paasivu/alisivu1/ali2sivu1/ali3sivu1/',
                content: '<p>Sisältö</p>',
              },
              {
                title: 'Alisivun1 alisivu1 alisivu2',
                uri: '/paasivu/alisivu1/ali2sivu1/ali3sivu2/',
              },
              {
                title: 'Alisivun1 alisivu1 alisivu3',
                uri: '/paasivu/alisivu1/ali2sivu1/ali3sivu3/',
              },
            ],
          },
          {
            title: 'Alisivu1 alisivu2',
            uri: '/paasivu/alisivu1/ali2sivu2/',
          },
          {
            title: 'Alisivu1 alisivu3',
            uri: '/paasivu/alisivu1/ali2sivu3/',
          },
        ],
      },
      learningMaterialsPage,
      {
        title: 'Alisivu3',
        uri: '/paasivu/alisivu3/',
      },
      {
        title: 'Alisivu4',
        uri: '/paasivu/alisivu4/',
      },
    ],
  },
];

function initializeMocks(pageHierarchy: PageHierarchy[]) {
  const mocks = {
    Page: [],
    SubPagesSearch: [
      {
        variables: {
          first: 10,
          id: '/paasivu/oppimateriaalit/',
          idType: 'URI',
          search: '',
        },
        response: {
          page: {
            id: 'page',
            children: {
              pageInfo: {
                endCursor: 10,
                hasNextPage: false,
              },
              edges: searchablePages.map((p) => ({
                cursor: '',
                node: fakePage({ title: p.title, uri: p.uri }),
              })),
            },
          },
        },
      },
    ],
  };
  function addMock(page: PageHierarchy) {
    mocks.Page.push({
      variables: {
        id: normalizeCmsUri(page.uri),
        idType: PageIdType.Uri,
      },
      response: {
        page: fakePage({
          title: page.title,
          uri: page.uri,
          content: page.content,
          children: {
            nodes: page.children
              ? page.children.map((child) =>
                  fakePage({
                    uri: child.uri,
                    title: child.title,
                    content: child.content,
                  })
                )
              : [],
          },
        }),
      },
    });
    page.children?.forEach((child) => addMock(child));
  }
  pageHierarchy.forEach((page) => addMock(page));
  return mocks;
}

const mocks = initializeMocks(pageHierarchy);

beforeEach(() => {
  server.use(
    graphql.query('Page', (req, res, ctx) => {
      const mock = mocks.Page.find(({ variables }) => {
        return dequal(variables, req.variables);
      });

      if (mock) {
        return res(ctx.data(mock.response));
      }

      // eslint-disable-next-line no-console
      console.error('Missing mock with variables: ', req.variables);
      return res(
        ctx.errors([
          {
            message: `Missing mock for the following query: \n\r\n\rquery: Page \n\rvariables: ${JSON.stringify(
              req.variables,
              null,
              2
            )}`,
          },
        ])
      );
    }),
    graphql.query('SubPagesSearch', (req, res, ctx) => {
      const mock = mocks.SubPagesSearch.find(({ variables }) => {
        return dequal(variables, req.variables);
      });

      if (mock) {
        return res(ctx.data(mock.response));
      }

      // eslint-disable-next-line no-console
      console.error('Missing mock with variables: ', req.variables);
      return res(
        ctx.errors([
          {
            message: `Missing mock for the following query: \n\r\n\rquery: Page \n\rvariables: ${JSON.stringify(
              req.variables,
              null,
              2
            )}`,
          },
        ])
      );
    })
  );
});

const getBreadcrumbsContainer = () =>
  within(screen.getByTestId(breadcrumbsContainerTestId));
const getCmsNavigationContainer = () =>
  within(screen.getByTestId(cmsNavigationContainerTestId));

test('renders CMS page and navigation flow works', async () => {
  const { container } = renderWithRoute(<CmsPage />, {
    routes: [`/fi${ROUTES.CMS_PAGE.replace(':slug', 'paasivu')}`],
    path: `/fi${ROUTES.CMS_PAGE}+`,
  });
  await wait();
  await testLevel1Navigation();
  await testLevel2Navigation();
  await testLevel3Navigation();
  await testLevel4Navigation();
  await testBreadcrumbNavigation();

  async function testLevel1Navigation() {
    // all level 1 navigation links should be rendered
    pageHierarchy[0].children.forEach(({ title }) => screen.getByText(title));

    // level 2 navigation links shouldn't be rendered yet
    pageHierarchy[0].children[0].children.forEach(({ title }) =>
      expect(screen.queryByText(title)).not.toBeInTheDocument()
    );

    // level 1 navigation breadcrumbs
    const breadcrumbs = getBreadcrumbsContainer();
    breadcrumbs.getByRole('link', { name: /kotisivu/i });
    breadcrumbs.getByText('Pääsivu');

    // main page title and content should be rendered
    screen.getByRole('heading', { name: mainPageTitle });
    screen.getByText(mainPageContent);
  }

  async function testLevel2Navigation() {
    userEvent.click(await screen.findByText(/alisivu1/i));
    await wait();

    // all level 1 and 2 navigation links shuld be rendered
    pageHierarchy[0].children.forEach(({ title }) =>
      getCmsNavigationContainer().getByText(title)
    );
    pageHierarchy[0].children[0].children.forEach(({ title }) =>
      getCmsNavigationContainer().getByText(title)
    );

    // first child should be active after navigating to it
    const firstNavigationChild = screen.getByRole('link', {
      name: pageHierarchy[0].children[0].title,
    });
    expect(firstNavigationChild).toHaveClass('activeLink');

    // level 2 navigation breadcrumbs
    const breadcrumbs = getBreadcrumbsContainer();
    breadcrumbs.getByRole('link', { name: 'Kotisivu' });
    breadcrumbs.getByRole('link', { name: 'Pääsivu' });
    breadcrumbs.getByText('Alisivu1');

    // sub page title and content should be rendered
    screen.getByRole('heading', { name: subPageTitle });
    screen.getByText(subPageContent);
  }

  async function testLevel3Navigation() {
    userEvent.click(await screen.findByText(/alisivu1 alisivu1/i));
    await wait();

    // all level 1 and 2 and 3  navigation links shuld be rendered
    pageHierarchy[0].children.forEach(({ title }) =>
      getCmsNavigationContainer().getByText(title)
    );
    pageHierarchy[0].children[0].children.forEach(({ title }) =>
      getCmsNavigationContainer().getByText(title)
    );
    pageHierarchy[0].children[0].children[0].children.forEach(({ title }) =>
      getCmsNavigationContainer().getByText(title)
    );

    const navigation = getCmsNavigationContainer();
    // two navigation links should be active after navigating
    const firstNavigationChild = navigation.getByRole('link', {
      name: pageHierarchy[0].children[0].title,
    });
    const secondNavigationChild = navigation.getByRole('link', {
      name: pageHierarchy[0].children[0].children[0].title,
    });
    expect(firstNavigationChild).toHaveClass('activeLink');
    expect(secondNavigationChild).toHaveClass('activeLink');

    // level 3 navigation breadcrumbs
    const breadcrumbs3 = getBreadcrumbsContainer();
    breadcrumbs3.getByRole('link', { name: 'Kotisivu' });
    breadcrumbs3.getByRole('link', { name: 'Pääsivu' });
    breadcrumbs3.getByRole('link', { name: 'Alisivu1' });
    breadcrumbs3.getByText('Alisivu1 alisivu1');
  }

  async function testLevel4Navigation() {
    userEvent.click(await screen.findByText(/alisivu1 alisivu1 alisivu1/i));
    await wait();

    // all level 1 and 2 and 3 navigation links should still be rendered
    pageHierarchy[0].children.forEach(({ title }) =>
      getCmsNavigationContainer().getByText(title)
    );
    pageHierarchy[0].children[0].children.forEach(({ title }) =>
      getCmsNavigationContainer().getByText(title)
    );
    pageHierarchy[0].children[0].children[0].children.forEach(({ title }) =>
      getCmsNavigationContainer().getByText(title)
    );

    // three navigation links should be active after navigating
    const navigation = getCmsNavigationContainer();
    const firstNavigationChild = navigation.getByRole('link', {
      name: pageHierarchy[0].children[0].title,
    });
    const secondNavigationChild = navigation.getByRole('link', {
      name: pageHierarchy[0].children[0].children[0].title,
    });
    const thirdNavigationChild = navigation.getByRole('link', {
      name: pageHierarchy[0].children[0].children[0].children[0].title,
    });
    expect(firstNavigationChild).toHaveClass('activeLink');
    expect(secondNavigationChild).toHaveClass('activeLink');
    expect(thirdNavigationChild).toHaveClass('activeLink');

    // level 4 navigation breadcrumbs
    const breadcrumbs = getBreadcrumbsContainer();
    breadcrumbs.getByRole('link', { name: 'Kotisivu' });
    breadcrumbs.getByRole('link', { name: 'Pääsivu' });
    breadcrumbs.getByRole('link', { name: 'Alisivu1' });
    breadcrumbs.getByRole('link', { name: 'Alisivu1 alisivu1' });
    breadcrumbs.getByText('Alisivu1 alisivu1 alisivu1');

    expect(container).toMatchSnapshot();
  }

  async function testBreadcrumbNavigation() {
    // Test breadcrumb navigation
    const breadcrumbs = getBreadcrumbsContainer();
    const breadcrumbLink = breadcrumbs.getByRole('link', {
      name: 'Alisivu1 alisivu1',
    });
    userEvent.click(breadcrumbLink);
    await waitFor(() => {
      expect(
        breadcrumbs.queryByText('Alisivu1 alisivu1 alisivu1')
      ).not.toBeInTheDocument();
    });
    // should be as link in the document because it is the current page
    expect(
      breadcrumbs.queryByRole('link', { name: 'Alisivu1 alisivu1' })
    ).not.toBeInTheDocument();
    // but text should be found as it is the current page
    breadcrumbs.getByText('Alisivu1 alisivu1');
  }
});

test('CMS sub pages can be searched', async () => {
  const { history } = renderWithRoute(<CmsPage />, {
    routes: [`/fi${ROUTES.CMS_PAGE.replace(':slug', 'paasivu')}`],
    path: `/fi${ROUTES.CMS_PAGE}+`,
  });

  userEvent.click(await screen.findByRole('link', { name: 'Oppimateriaalit' }));

  await screen.findByRole('heading', { name: /oppimateriaalit/i });
  await screen.findByText(/kaikki opimateriaalit/i);
  await screen.findByRole('textbox', { name: /haku/i });

  for (const page of searchablePages) {
    await screen.findByRole('heading', { name: page.title });
  }

  // navigation works from page card
  userEvent.click(
    screen.getByRole('link', {
      name: /oppimateriaali1/i,
    })
  );

  await wait();

  const breadcrumbs = getBreadcrumbsContainer();
  breadcrumbs.getByText('Oppimateriaali1');
  screen.getByRole('heading', { name: 'Oppimateriaali1' });

  expect(history.location.pathname).toBe(
    '/fi/cms-page/paasivu/oppimateriaalit/oppimateriaali1/'
  );
});
