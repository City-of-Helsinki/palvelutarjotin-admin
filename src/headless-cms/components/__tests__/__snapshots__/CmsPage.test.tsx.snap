// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CMS page and navigation flow works 1`] = `
<div>
  <div>
    <nav
      data-testid="cms-navigation-container"
    >
      <div
        class="navigationRowContainer"
      >
        <div
          class="container default container"
        >
          <div
            class="navigationRow"
          >
            <a
              class="activeLink"
              href="/fi/cms-page/paasivu/alisivu1/"
            >
              <span>
                Alisivu1
              </span>
            </a>
            <a
              class=""
              href="/fi/cms-page/paasivu/alisivu3/"
            >
              <span>
                Alisivu3
              </span>
            </a>
            <a
              class=""
              href="/fi/cms-page/paasivu/alisivu4/"
            >
              <span>
                Alisivu4
              </span>
            </a>
            <a
              class=""
              href="/fi/cms-page/paasivu/oppimateriaalit/"
            >
              <span>
                Oppimateriaalit
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="navigationRowContainer"
      >
        <div
          class="container default container"
        >
          <div
            class="navigationRow"
          >
            <a
              class="activeLink"
              href="/fi/cms-page/paasivu/alisivu1/ali2sivu1/"
            >
              <span>
                Alisivu1 alisivu1
              </span>
            </a>
            <a
              class=""
              href="/fi/cms-page/paasivu/alisivu1/ali2sivu2/"
            >
              <span>
                Alisivu1 alisivu2
              </span>
            </a>
            <a
              class=""
              href="/fi/cms-page/paasivu/alisivu1/ali2sivu3/"
            >
              <span>
                Alisivu1 alisivu3
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="navigationRowContainer"
      >
        <div
          class="container default container"
        >
          <div
            class="navigationRow"
          >
            <a
              class="activeLink"
              href="/fi/cms-page/paasivu/alisivu1/ali2sivu1/ali3sivu1/"
            >
              <span>
                Alisivu1 alisivu1 alisivu1
              </span>
            </a>
            <a
              class=""
              href="/fi/cms-page/paasivu/alisivu1/ali2sivu1/ali3sivu2/"
            >
              <span>
                Alisivun1 alisivu1 alisivu2
              </span>
            </a>
            <a
              class=""
              href="/fi/cms-page/paasivu/alisivu1/ali2sivu1/ali3sivu3/"
            >
              <span>
                Alisivun1 alisivu1 alisivu3
              </span>
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container default container"
    >
      <ul
        class="breadcrumbList"
        data-testid="breadcrumbs-container"
      >
        <li>
          <a
            href="/"
          >
            Kotisivu
          </a>
        </li>
        <li>
          <a
            href="/fi/cms-page/paasivu/"
          >
            Pääsivu
          </a>
        </li>
        <li>
          <a
            href="/fi/cms-page/paasivu/alisivu1/"
          >
            Alisivu1
          </a>
        </li>
        <li>
          <a
            href="/fi/cms-page/paasivu/alisivu1/ali2sivu1/"
          >
            Alisivu1 alisivu1
          </a>
        </li>
        <li>
          Alisivu1 alisivu1 alisivu1
        </li>
      </ul>
    </div>
    <div
      class="container default"
    >
      <h1>
        Alisivu1 alisivu1 alisivu1
      </h1>
      <p>
        Sisältö
      </p>
    </div>
  </div>
</div>
`;
